sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -122,8 +122,8 @@ object StringUtils extends Logging {
           val available = maxLength - length
           val stringToAppend = if (available >= sLen) s else s.substring(0, available)
           strings.append(stringToAppend)
+          length += sLen
         }
-        length += sLen
```

Comment:
How about:
```scala
        length = Math.min(length.toLong + sLen, ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH).toInt
```

Reply:
@MaxGekk Yeah.. Sure. Just wondering we absolutely need to change the length in case we are not doing an append. The error message you are referring to, is this captured in some test ? I ran catalyst and sql tests and didn't see any failure.

Reply:
> The error message you are referring to, is this captured in some test ?

Looks like not. Could you add a test for that?

Reply:
@MaxGekk Sure.. Will add a test. Can you please point me to the error message please ? The link you have, when i click .. its not taking me to the relevant code ?

Reply:
In the child class, here https://github.com/apache/spark/blob/d9b30694122f8716d3acb448638ef1e2b96ebc7a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala#L156
https://github.com/apache/spark/blob/d9b30694122f8716d3acb448638ef1e2b96ebc7a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala#L151

Reply:
Nitpicking, but:
Why not only update length inside the `if (!atLimit)` block as originally shown?
And only add `stringToAppend.length()` there, if the point of `Math.min()` is to avoid overflow if adding a massive string where only part was added.

Reply:
@srowen I think, the original intent of keeping this length outside the if block (which i missed and @MaxGekk pointed out) is to keep the true length of the input. So when we are producing a SQL plan, we want to tell users how much was that the size of the explain string and how much we truncated based on max length specification. So caller could call append method even after we have gone past the max. 
cc @MaxGekk

Reply:
Oh I see. If overflow is an issue, then make it a long? Capping it at this constant would then seem to potentially underreport the actual appended total.
Maybe that creates other problems, not sure.

Reply:
@srowen You are right that, we may under report when the length becomes more than MAX_ROUNDED_ARRAY_LENGTH. I did consider changing the message to cover the case to say something like "greater than or equal to N". But thought against it as we are adding code to cover may be a extremely corner case. Please let me know if you have any suggestion here.

About changing to long, even if we did it, we can in theory hit the long limit, right ?

Reply:
I'm not sure you can overflow a long - if 2^31-1 strings (max length of the array tracking them) are added, of length 2^31-1, you still don't overflow.
It's probably not worth debating so this is fine but I think a simpler change was also fine.

Reply:
Thank you @srowen.

Reply:
Ah, could you leave some comments about the intent here? I think its a bit hard to tell it at a glance.